### PR TITLE
Document benchmark GitHub Actions workflow, and update contribution guide

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR adds the benchmark GitHub Actions workflow for both testing and benchmarking, and updates contribution guide:

Screen Shot 2021-04-05 at 4 08 36 PM
Screen Shot 2021-04-05 at 4 08 23 PM

### CODE REVIEW

**Review by maropu — APPROVED** [review]
Looks nice and I wanna to use the new feature for my work.

**Comment by maropu** [comment]
btw, we don't update the PR template, too?

**Comment by HyukjinKwon** [comment]
I think it's okay ... I just updated the contribution guide instead :-).

**Review by wangyum — APPROVED** [review]

File: developer-tools.md [hunk 1]
```diff
@@ -216,25 +216,45 @@ In case of a failure the POD logs (driver and executors) can be found at the end
...
 
-<h3>Running tests in your forked repository using GitHub Actions</h3>
+<h3>Testing with GitHub Actions workflow</h3>
 
-GitHub Actions is a functionality within GitHub that enables continuous integration and a wide range of automation.
-We already have started using some action scripts and one of them is to run tests for [pull requests](https://spark.apache.org/contributing.html).
-If you are planning to create a new pull request, it is important to check if tests can pass on your branch before creating a pull request.
-This is because our GitHub Acrions script automatically runs tests for your pull request/following commits and
-this can burden our limited resources of GitHub Actions.
+Apache Spark leverages GitHub Actions that enables continuous integration and a wide range of automation. Apache Spark repository provides several GitHub Actions workflows for developers to run before creating a pull request.
```
Comment (MaxGekk) [inline developer-tools.md:221]: enables -> enable?

File: site/developer-tools.html [hunk 2]
```diff
@@ -394,27 +394,51 @@ <h4>Testing K8S</h4>
...
 
-<h3>Running tests in your forked repository using GitHub Actions</h3>
+<h3>Testing with GitHub Actions workflow</h3>
 
-<p>GitHub Actions is a functionality within GitHub that enables continuous integration and a wide range of automation.
-We already have started using some action scripts and one of them is to run tests for <a href="https://spark.apache.org/contributing.html">pull requests</a>.
-If you are planning to create a new pull request, it is important to check if tests can pass on your branch before creating a pull request.
-This is because our GitHub Acrions script automatically runs tests for your pull request/following commits and
-this can burden our limited resources of GitHub Actions.</p>
+<p>Apache Spark leverages GitHub Actions that enables continuous integration and a wide range of automation. Apache Spark repository provides several GitHub Actions workflows for developers to run before creating a pull request.</p>
```
Comment (MaxGekk) [inline site/developer-tools.html:399]: enables -> enable

**Review by MaxGekk — APPROVED** [review]

File: developer-tools.md (hunk 1 above)
Comment (HyukjinKwon) [reply developer-tools.md:221]: I believe "GitHub Actions" is considered as an uncountable noun (?).

Comment (HyukjinKwon) [reply developer-tools.md:221]: e.g.) https://github.com/features/actions
> GitHub Actions makes

**Comment by HyukjinKwon** [comment]
Merged!

File: developer-tools.md (hunk 1 above)
Comment (srowen) [reply developer-tools.md:221]: I think it simply functions as singular despite "Actions". Pedantically "Github Actions, which enables" would make that smoother but it doesn't matter.

**Pull request closed without merging**